User can edit a listing#15

damwhit opened the pull request:
A user can edit a listing on both the listing show and index pages.  Jquery has been implemented correctly and has been tested in the feature.

damwhit committed:
renders 404 if there is no user

if a non logged in user hard codes /events or /sell a 404 will render

damwhit committed:
Merge branch 'master' into user-can-edit-a-listing#15

damwhit committed:
links to destroy and update working in listings index

spiked the listings index and now the links to update and destroy listings are working

damwhit committed:
user can edit a listing

show page is now dynamic, need to figure out seat addition and subtraction

damwhit committed:
show page is now dynamic

damwhit committed:
Merge branch 'master' into user-can-edit-a-listing#15

damwhit committed:
user can now successfully update a listing

changed the ui for editing a ticket listing.  Listing index page is now set up in such a way that seats can be edited correctly

damwhit committed:
user can edit a listing

test is passing for feature

erinnachen commented on Gemfile:
```diff
@@ -21,6 +21,7 @@ gem 'chartkick'
 gem 'groupdate'
 gem 'will_paginate', '~> 3.0.6'
 gem 'bootstrap-will_paginate'
+gem 'annotate'
```
cool addition.

erinnachen commented on app/controllers/listings_controller.rb:
```diff
@@ -22,11 +33,34 @@ def create
   end
 
   def index
+    require_user
```
this should probably be removed from the method and made a before action
before_action :require_user, only: [:index, :show]

erinnachen commented on app/controllers/listings_controller.rb:
```diff
@@ -22,11 +33,34 @@ def create
   end
 
   def show
+    require_user
```
see comment above

damwhit committed:
jquery almost working

can append rows correctly, however delete row button does not work on appended rows

damwhit committed:
show page is now good for listing editing

damwhit committed:
closer but still not able to delete populated rows

damwhit committed:
user can edit a listing on both show and index pages

Will probably want to eventually mess with the styling but a user is now able to edit a listing on both the listings index page as well as the show page for a specific listing.

damwhit committed:
user can edit a ticket listing

jquery is now tested for both add and remove seat buttons in edit form

damwhit committed:
fixed merge conflicts

damwhit committed:
hound

damwhit committed:
edit listing form is behaving correctly

rows are now appending only to the correct table

damwhit committed:
Merge branch 'master' into user-can-edit-a-listing#15

damwhit committed:
User can now edit a listing.  Jquery implementation is working correctly.

erinnachen commented:
Yay! Good work +++
